src/utils/worker/createRequestListener.ts

```diff
@@ -31,71 +29,44 @@ export const createRequestListener = (
 
     try {
       const request = parseWorkerRequest(message.payload)
-      context.emitter.emit('request:start', request)
-
-      // Set document cookies on the request.
-      setRequestCookies(request)
-
-      const {
-        response,
-        handler,
-        publicRequest,
-        parsedRequest,
-      } = await getResponse(request, context.requestHandlers)
-
-      // Handle a scenario when there is no request handler
-      // found for a given request.
-      if (!handler) {
-        onUnhandledRequest(
-          request,
-          context.requestHandlers,
-          options.onUnhandledRequest,
-        )
-        context.emitter.emit('request:unhandled', request)
-        context.emitter.emit('request:end', request)
-
-        return channel.send({ type: 'MOCK_NOT_FOUND' })
-      }
-
-      context.emitter.emit('request:match', request)
-
-      // Handle a scenario when there is a request handler,
-      // but it doesn't return any mocked response.
-      if (!response) {
-        console.warn(
-          '[MSW] Expected a mocking resolver function to return a mocked response Object, but got: %s. Original response is going to be used instead.',
-          response,
-        )
-
-        context.emitter.emit('request:end', request)
-
-        return channel.send({ type: 'MOCK_NOT_FOUND' })
-      }
-
-      readResponseCookies(request, response)
-
-      const responseWithSerializedHeaders: ResponseWithSerializedHeaders = {
-        ...response,
-        headers: headersToList(response.headers),
-      }
-
-      if (!options.quiet) {
-        setTimeout(() => {
-          handler.log(
-            publicRequest,
-            responseWithSerializedHeaders,
-            handler,
-            parsedRequest,
-          )
-        }, response.delay)
-      }
-
-      context.emitter.emit('request:end', request)
-
-      channel.send({
-        type: 'MOCK_SUCCESS',
-        payload: responseWithSerializedHeaders,
-      })
+      await handleRequest<ResponseWithSerializedHeaders>(
+        request,
+        context.requestHandlers,
+        options,
+        context.emitter,
+        {
+          transformResponse(response) {
+            return {
+              ...response,
+              headers: headersToList(response.headers),
+            }
+          },
+          onBypassResponse() {
+            return channel.send({
+              type: 'MOCK_NOT_FOUND',
+            })
+          },
+          onMockedResponse(response) {
+            channel.send({
```

Comment:
It's important to signal the worker _ignoring the response's delay_, as the worker resolves the response respecting that delay. Otherwise, the delay would get doubled.